pkg/template/api/validation/validation.go

```diff
@@ -33,6 +40,11 @@ func ValidateProcessedTemplate(template *api.Template) fielderrors.ValidationErr
 func ValidateTemplate(template *api.Template) (errs fielderrors.ValidationErrorList) {
 	errs = validation.ValidateObjectMeta(&template.ObjectMeta, true, validation.ValidatePodName).Prefix("metadata")
 	errs = append(errs, validateTemplateBody(template)...)
+	fields := map[string]string{}
+	for i := range template.Parameters {
+		fields[template.Parameters[i].Name] = template.Parameters[i].Description
+	}
+	errs = append(errs, validation.ValidateAnnotations(fields, "description")...)
```

Comment:
as long as the substitution code handles all the values the annotation key validation allows, that's fine